assets/dev/js/editor/document/dynamic/commands/settings.js

```diff
@@ -0,0 +1,69 @@
+import Base from '../../commands/base';
+import Settings from '../../elements/commands/settings';
+
+export default class extends Base {
+	static restore( historyItem, isRedo ) {
+		const data = historyItem.get( 'data' );
+
+		historyItem.get( 'containers' ).forEach( ( container ) => {
+			const changes = data.changes[ container.id ];
+
+			$e.run( 'document/dynamic/settings', {
+				container,
+				settings: isRedo ? changes.new : changes.old,
+			} );
+
+			container.panel.refresh();
+		} );
+	}
+
+	validateArgs( args ) {
+		this.requireContainer( args );
+
+		this.requireArgumentConstructor( 'settings', Object, args );
+	}
+
+	getHistory( args ) {
+		const { settings, containers = [ args.container ] } = args,
+			changes = {};
+
+		containers.forEach( ( container ) => {
+			const { id } = container;
+
+			if ( ! changes[ id ] ) {
+				changes[ id ] = {};
+			}
+
+			changes[ id ] = {
+				old: container.dynamic.toJSON(),
+				new: settings,
+			};
+		} );
+
+		const subTitle = Settings.getSubTitle( args );
+
+		return {
+			containers,
+			subTitle,
+			data: { changes },
+			type: 'change',
+			restore: this.constructor.restore,
+		};
+	}
+
+	apply( args ) {
+		const { settings, containers = [ args.container ] } = args;
+
+		containers.forEach( ( container ) => {
```

Comment:
since 'document/elements/settings' starts log and trigger `control-behavior::setDynamicValue` and that calls to `document/dynamic/settings` it is not possible.

currently its does two times startLog the second startLog ( of dynamic ) goes as subItem.